src/components/token/token.tsx

```diff
@@ -29,95 +40,137 @@ const shapeToClassMap: { [shape in TokenShape]: string } = {
+  euiColorVis9: 'euiToken--euiColorVis9',
+  gray: 'euiToken--gray',
 };
 
 export const COLORS = keysOf(colorToClassMap);
 
-interface TokenProps {
+export interface TokenProps {
   /**
    * An EUI icon type
    */
   iconType: IconType;
   /**
-   * Size of the token
+   * For best results use one of the vis color names (or 'gray').
+   * Or supply your own color (can be used with dark or no fill only).
    */
-  size?: TokenSize;
+  color?: TokenColor | string;
+  /**
+   * Outer shape surrounding the icon
+   */
+  shape?: TokenShape;
+  /**
+   * `light` for lightened color with border, `dark` for solid, or `none`
+   */
+  fill?: TokenFill;
   /**
-   * By default EUI will auto color tokens. You can can however control it
-   * - `color`: can be `tokenTint01` thru `tokenTint10`
-   * - `shape`: square, circle, rectangle as options
-   * - `fill`: makes it a solid color
-   * - `hideBorder`: disables the outer border
+   * Size of the token
    */
-  displayOptions?: EuiTokenMapDisplayOptions;
+  size?: TokenSize;
+  style?: CSSProperties;
 }
 
 export type EuiTokenProps = CommonProps &
   TokenProps &
-  HTMLAttributes<HTMLDivElement>;
+  HTMLAttributes<HTMLSpanElement>;
 
 export const EuiToken: FunctionComponent<EuiTokenProps> = ({
   iconType,
-  displayOptions = {},
+  color,
+  fill,
+  shape,
   size = 's',
+  style = {},
   className,
   ...rest
 }) => {
-  // Check if display options is empty
-  const displayOptionsIsEmpty =
-    Object.keys(displayOptions).length === 0 &&
-    displayOptions.constructor === Object;
-
-  let tokenShape: TokenShape;
-  let tokenColor: TokenColor;
-  let fill: boolean;
-  let tokenHidesBorder: boolean;
-
-  // Check if this has a mapping, and doesn't have custom displayOptions
-  if (iconType in TOKEN_MAP && displayOptionsIsEmpty) {
-    const mapping = TOKEN_MAP[iconType as EuiTokenMapType];
-    // These should be found in the standard mappings, but apply defaults
-    // just in case.
-    tokenShape = mapping.shape || 'square';
-    tokenColor = mapping.color || 'tokenTint01';
-    fill = mapping.fill ? true : false;
-    tokenHidesBorder = mapping.hideBorder ? true : false;
+  // Set the icon size to the same as the passed size
+  // unless they passed `xs` which IconSize doesn't support
+  let finalSize: IconSize = size === 'xs' ? 's' : size;
+
+  // When displaying at the small size, the token specific icons
+  // should actually be displayed at medium size
+  if (String(iconType).indexOf('token') === 0 && size === 's') {
+    finalSize = 'm';
+  }
+
+  const defaultDisplay = {
+    color: 'gray',
+    fill: 'light',
+    shape: 'sqaure',
+  };
+  const currentDisplay = {
+    color,
+    fill,
+    shape,
+  };
+  let finalDisplay;
+
+  // If the iconType passed is one of the prefab token types,
+  // grab its properties
+  if (iconType in TOKEN_MAP) {
+    const tokenDisplay = TOKEN_MAP[iconType as EuiTokenMapType];
+    finalDisplay = defaults(currentDisplay, tokenDisplay, defaultDisplay);
```

Comment:
We can avoid the lodash dependency with,

```suggestion
    finalDisplay = { ...defaultDisplay, ...tokenDisplay, ...currentDisplay };
```

(order of operations are reversed from `defaults`, properties on a right-most object will override those defined earlier)

Reply:
_I realized I actually don't need the `defaultDisplay` as it's duplicative of the `finalColor = finalDisplay.color || 'gray';` etc checks below_... But given that:

This actually doesn't seem to work the same way as you mentioned.

Given something like:

```tsx
currentDisplay = {
    color: undefined,
    fill: undefined,
    shape: undefined,
};

tokenDisplay = {
    color: 'blue',
    fill: 'light',
    shape: 'square',
}
```

Then when using the spread notation like:

```tsx
finalDisplay = { ...tokenDisplay, ...currentDisplay };
```

The result is

```tsx
currentDisplay = {
    color: undefined,
    fill: undefined,
    shape: undefined,
};
```

Since it gets overridden. Whereas, the lodash function doesn't replace actual values with null/undefined.

Reply:
That's a good catch.